olive/systems/python_environment/python_environment_system.py

```diff
@@ -0,0 +1,261 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import logging
+import os
+import pickle
+import tempfile
+from copy import deepcopy
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+import torch
+
+from olive.common.utils import run_subprocess
+from olive.evaluator.evaluation import compute_accuracy, compute_latency, format_onnx_input, get_user_config
+from olive.evaluator.metric import Metric, MetricType
+from olive.model import OliveModel, ONNXModel
+from olive.passes.olive_pass import Pass
+from olive.systems.common import Device, SystemType
+from olive.systems.olive_system import OliveSystem
+from olive.systems.system_config import PythonEnvironmentTargetUserConfig
+
+logger = logging.getLogger(__name__)
+
+
+class PythonEnvironmentSystem(OliveSystem):
```

Comment:
>PythonEnvironmentSystem [](http://example.com/codeflow?start=6&length=23)

should VirtualEnv be clear?

Reply:
Technically this could be a conda environment. Or just any environment that has a python executable.